### PR TITLE
testing/e2e: add auth method pagination tests

### PR DESCRIPTION
**Pull request by mikemountain**

### CODE REVIEW

File: testing/internal/e2e/tests/base/auth_method_password_test.go [hunk 1]
```diff
@@ -136,3 +140,176 @@ func TestCliAuthMethodPassword(t *testing.T) {
 	)
 	require.NoError(t, output.Err, string(output.Stderr))
 }
+
+// TestCliPaginateAuthMethods asserts that the CLI autamatically paginates to retrieve
```
Comment (johanbrandhorst) [inline testing/internal/e2e/tests/base/auth_method_password_test.go:144]:
```suggestion
// TestCliPaginateAuthMethods asserts that the CLI automatically paginates to retrieve
```

File: testing/internal/e2e/tests/base/auth_method_password_test.go [hunk 2]
```diff
@@ -136,3 +140,176 @@ func TestCliAuthMethodPassword(t *testing.T) {
...
+		ctx := context.Background()
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("scopes", "delete", "-id", newOrgId))
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+
+	// Create enough auth methods to overflow a single page.
+	// Three auth methods are created by default, so we
+	// need to remove them from the total auth method count.
+	numPrecreatedAuthMethods := 3
```
Comment (johanbrandhorst) [inline testing/internal/e2e/tests/base/auth_method_password_test.go:166]: Are there? It seems to not be the case in the API test below.

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]

File: testing/internal/e2e/tests/base/auth_method_password_test.go [hunk 3]
```diff
@@ -136,3 +140,176 @@ func TestCliAuthMethodPassword(t *testing.T) {
...
+	var initialAuthMethods authmethods.AuthMethodListResult
+	err = json.Unmarshal(output.Stdout, &initialAuthMethods)
+	require.NoError(t, err)
+
+	var returnedIds []string
+	// Ignore the precreated auth methods, which will appear at the end
+	for _, authMethod := range initialAuthMethods.Items[:len(initialAuthMethods.Items)-numPrecreatedAuthMethods] {
+		returnedIds = append(returnedIds, authMethod.Id)
+	}
+	require.Len(t, initialAuthMethods.Items, c.MaxPageSize+1)
```
Comment (moduli) [inline testing/internal/e2e/tests/base/auth_method_password_test.go:191]: Looks like the test is currently failing here
```
should have 11 item(s), but has 9
```

**Review by moduli — CHANGES_REQUESTED** [review]

**Review by johanbrandhorst — APPROVED** [review]
LGTM if the tests pass

File: testing/internal/e2e/tests/base/auth_method_password_test.go (hunk 3 above)
Comment (moduli) [reply testing/internal/e2e/tests/base/auth_method_password_test.go:188]: It has been resolved now

File: testing/internal/e2e/tests/base/auth_method_password_test.go [hunk 4]
```diff
@@ -136,3 +140,174 @@ func TestCliAuthMethodPassword(t *testing.T) {
...
+		authMethodId := boundary.CreateNewAuthMethodApi(t, ctx, client, newOrgId)
+		authMethodIds = append(authMethodIds, authMethodId)
+	}
+
+	// List auth methods
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"auth-methods", "list",
+			"-scope-id", newOrgId,
+			"-recursive",
```
Comment (moduli) [inline testing/internal/e2e/tests/base/auth_method_password_test.go:175]: Does this need the `-recursive` flag?

File: testing/internal/e2e/tests/base/auth_method_password_test.go [hunk 5]
```diff
@@ -136,3 +140,174 @@ func TestCliAuthMethodPassword(t *testing.T) {
...
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// List auth methods again, should have new one but not old one
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"auth-methods", "list",
+			"-scope-id", newOrgId,
+			"-recursive",
```
Comment (moduli) [inline testing/internal/e2e/tests/base/auth_method_password_test.go:209]: Does this need the `-recursive` flag?

File: testing/internal/e2e/tests/base/auth_method_password_test.go [hunk 6]
```diff
@@ -136,3 +140,174 @@ func TestCliAuthMethodPassword(t *testing.T) {
...
+	})
+
+	// Create enough auth methods to overflow a single page.
+	var authMethodIds []string
+	for i := 0; i < c.MaxPageSize+1; i++ {
+		authMethodIds = append(authMethodIds, boundary.CreateNewAuthMethodApi(t, ctx, client, newOrgId))
+	}
+
+	// List auth methods
+	initialAuthMethods, err := amClient.List(ctx, newOrgId, authmethods.WithRecursive(true))
```
Comment (moduli) [inline testing/internal/e2e/tests/base/auth_method_password_test.go:261]: Does this need the `WithRecursive` option>

File: testing/internal/e2e/tests/base/auth_method_password_test.go [hunk 7]
```diff
@@ -136,3 +140,174 @@ func TestCliAuthMethodPassword(t *testing.T) {
...
+	require.True(t, ok)
+	assert.Len(t, mapSliceItems, c.MaxPageSize+1)
+
+	// Create a new auth method and destroy one of the other auth methods
+	newAuthMethodId := boundary.CreateNewAuthMethodApi(t, ctx, client, newOrgId)
+	_, err = amClient.Delete(ctx, initialAuthMethods.Items[0].Id)
+	require.NoError(t, err)
+
+	// List auth methods again, should have new one but not old one
+	newAuthMethods, err := amClient.List(ctx, newOrgId, authmethods.WithListToken(initialAuthMethods.ListToken), authmethods.WithRecursive(true))
```
Comment (moduli) [inline testing/internal/e2e/tests/base/auth_method_password_test.go:286]: Does this need the `WithRecursive` option?

**Pull request merged**